### ISSUE TITLE
Improvement : Send 403 on denied

### ISSUE THREAD

**Issue by LtSich** [crowdsecurity/cs-nginx-bouncer#62]
Hi,

Can you make that nginx sends a 403 response when the bouncer denies access? 

Like this access log : 
xxx.xxx.xxx.xxx - - [15/Oct/2024:09:09:26 +0200] "GET /log.php HTTP/2.0" 200 5620 "-" "-"

But the error log : 
2024/10/15 09:09:26 [alert] 515076#515076: *6141 [lua] crowdsec.lua:748: Allow(): [Crowdsec] denied 'xxx.xxx.xxx.xxx' with 'captcha', client: xxx.xxx.xxx.xxx, server: tld.com, request: "GET /log.php HTTP/2.0"

Nginx should respond with 403 and not 200... This is important to not cache response (like with Cloudflare)...

**Comment by LaurenceJJones** [comment]
> Hi,
> 
> Can you make that nginx sends a 403 response when the bouncer denies access?
> 
> Like this access log : xxx.xxx.xxx.xxx - - [15/Oct/2024:09:09:26 +0200] "GET /log.php HTTP/2.0" 200 5620 "-" "-"
> 
> But the error log : 2024/10/15 09:09:26 [alert] 515076#515076: *6141 [lua] crowdsec.lua:748: Allow(): [Crowdsec] denied 'xxx.xxx.xxx.xxx' with 'captcha', client: xxx.xxx.xxx.xxx, server: tld.com, request: "GET /log.php HTTP/2.0"
> 
> Nginx should respond with 403 and not 200... This is important to not cache response (like with Cloudflare)...

If the upstream proxies respects cache headers it should reject caching the captcha response:

https://github.com/crowdsecurity/lua-cs-bouncer/blob/2270a7e13a18dfa94bf5cc462d1d908a3c43358f/lib/crowdsec.lua#L728-L729

However, there can be a debate on the response code, however, in my view `403` be used when the IP is completely denied and cannot regain access by sending a authentication or a captcha solution.

```
This status is similar to [401](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401), except that for 403 Forbidden responses, authenticating or re-authenticating makes no difference.
```

**Comment by LtSich** [comment]
Yep, it's more when I choose to force cache on cloudflare (aka cache everything), you can only filter on response code, not on cache header... 
But 401 response code could be good too on my opinion.

For me it's more like a 200 response code is "everything is ok", but it's not... As the bouncer doesn't send the website content, just the captcha challenge... Or the block page...

**Comment by LaurenceJJones** [comment]
Classing as closed, once a new release has been issued this feature will be available.

Just so it known there a new option as `CAPTCHA_RET_CODE` which takes a int as `RET_CODE` does, which has to be within the supported return codes.

https://github.com/crowdsecurity/lua-cs-bouncer/blob/6f19b672bad0ad5cc679faf1a9f4221f78740071/lib/plugins/crowdsec/utils.lua#L7-L21
